### PR TITLE
8264948: Check for TLS extensions total length

### PR DESCRIPTION
**Pull request by XueleiFan**
To improve the readability, it would be nice to check the TLS extensions total length while parsing.

No new regression test,  trial update.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8264948](https://bugs.openjdk.java.net/browse/JDK-8264948): Check for TLS extensions total length

### Reviewers
 * [Jamil Nimeh](https://openjdk.java.net/census#jnimeh) (@jnimeh - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3405/head:pull/3405` \
`$ git checkout pull/3405`

Update a local copy of the PR: \
`$ git checkout pull/3405` \
`$ git pull https://git.openjdk.java.net/jdk pull/3405/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3405`

View PR using the GUI difftool: \
`$ git pr show -t 3405`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3405.diff">https://git.openjdk.java.net/jdk/pull/3405.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/sun/security/ssl/SSLExtensions.java [hunk 1]
```diff
@@ -54,7 +54,19 @@
...
+                    Alert.ILLEGAL_PARAMETER,
+                    "Incorrect extensions: no length field");
+        }
+
         int len = Record.getInt16(m);
+        if (len > m.remaining()) {
+            throw hm.handshakeContext.conContext.fatal(
+                    Alert.ILLEGAL_PARAMETER,
+                    "Insufficient extensions data");
+        }
```
Comment (jnimeh) [inline src/java.base/share/classes/sun/security/ssl/SSLExtensions.java:68]: For both of these blocks the checks themselves look OK, but illegal_parameter I thought was more for cases where a field value is out of range or inconsistent with already negotiated parameters.  I would think that decode_error would be more appropriate to cases like this where the encoding is structurally incorrect and the length doesn't match the actual data size.

Comment (XueleiFan) [reply src/java.base/share/classes/sun/security/ssl/SSLExtensions.java:68]: Good catch!  Updated to use decode_error.

**Review by jnimeh — APPROVED** [review]
Looks good to me.

**Pull request closed without merging**